bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,760 @@
+package org.roaringbitmap.bsi.longlong;
+
+import static org.roaringbitmap.longlong.Roaring64NavigableMap.SERIALIZATION_MODE_LEGACY;
+import static org.roaringbitmap.longlong.Roaring64NavigableMap.SERIALIZATION_MODE_PORTABLE;
+
+import org.roaringbitmap.bsi.BitmapSliceIndex;
+import org.roaringbitmap.bsi.Pair;
+import org.roaringbitmap.bsi.WritableUtils;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+/**
+ * Roaring64NavigableMap based bsi implementation.
+ * The default serialization format for underlying Roaring64NavigableMap is
+ * {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE}. And you can also switch to
+ * {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE} mode as you wish.
+ * NOTE: The serialization format is *not* compatible with bsi implementation of roaring(Go),
+ * @see <a href="https://github.com/RoaringBitmap/roaring/blob/b32ae1a8cc386adbf7753251b8a580ed37f270a9/roaring64/bsi64.go#L882">bsi64.WriteTo</a>
+ */
+public class Roaring64NavigableMapSliceIndex {
+  /**
+   * the maxValue of this bsi
+   */
+  private long maxValue;
+
+  /**
+   * the minValue of this bsi
+   */
+  private long minValue;
+
+  /**
+   * the bit component slice Array of this bsi
+   */
+  private Roaring64NavigableMap[] bA;
+
+  /**
+   * the exist bitmap of this bsi which means the columnId have value in this bsi
+   */
+  private Roaring64NavigableMap ebM;
+
+  private Boolean runOptimized = false;
+
+  /*
+   * The default serialization format is {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE}.
+   * all underlying 'Roaring64NavigableMap's should have the same serialization format
+   */
+  private int serializationMode = SERIALIZATION_MODE_PORTABLE;
```

Comment:
I feel weird BSI has its own default default serializationMode, not using the existing+customizable SERIALZITION_MODE.